### DISCUSSION TITLE
UX Improvement Ideas

### DISCUSSION THREAD

**Discussion by graciousgrey** [photoprism/photoprism#4264] [General]
Below you will find some initial feedback on the [UX report from arendtio](https://github.com/photoprism/photoprism/issues/3862#issuecomment-2028563657).

First of all, thank you very much for providing the report :heart: 

Most likely I won't be able to go into every point today, but I will do my best to continue when my time allows. 

**05** Based on what you describe, I'm assuming you have a static size limit of 2048 set. If that is the case and you want to display a larger image, the thumbnail will need to be created on demand. To solve this problem, you can increase the static size limit so that the larger thumbnails are also created during indexing. https://docs.photoprism.app/user-guide/settings/advanced/#dynamic-and-static-size-limits

**13**, **15**, **27**, We agree that the current photo viewer is not optimized for use with videos and live photos. We are therefore planning to develop our own viewer to fix these (and other) issues. You can find the corresponding ticket here:
-  #1307 

The new photo viewer will also have an optional sidebar for displaying metadata, which could also solve 25.

**25**, **03** Once we have the new photo viewer with a sidebar, we could add cross-links there. Related issues: 
- #1307
- #797 

We might be able to add a link from the date in the card view to a search for photos taken on the same date in one of the next releases, similar to the feature we already have for the [location](https://docs.photoprism.app/user-guide/organize/places/#open-photo-from-search-in-places).

**07** You can find a similar discussion here:
- #4200 

A running import/indexing and/or the selection of a random sort order (which is not officially supported) could cause this problem, as both can lead to changes/inconsistencies in the results. If we can reproduce that this problem also occurs when no indexing is running and no random sort order is set, we will be happy to look into how we can fix it.

**12** We would love to hear more opinions on this topic. Changing the icon shouldn't be a big deal. However, the icon you suggested looks very similar to the icon we use for stacks and there is no variant for pausing the slideshow. If we find a more suitable icon (ideally a Material Design icon) that has a variant for play and pause, we could change it.

I could not reproduce **31**, **32**. In the edit dialog it is only possible to reject a name or add a name to a face. It is not possible to rename a person. It is possible to rename a person under People/Recognized. If you try to rename a person there to a name that is already used for another person, a dialog will appear in which you must confirm that you want to merge the two persons.

**30** We will take care of this as soon as OpenID Connect Support has been implemented. 
- #3124

**18** Advanced options for sharing links are planned for the future.  
- #557

**Comment by arendtio** [comment]
Hi @graciousgrey 
Thanks for starting this discussion; some hints sound very promising. I just wanted to let you know that I am on vacation at the moment, so it might take a week or two before I can take a closer look at all the points.

**Reply by graciousgrey** [reply]
Enjoy your vacation :)

**Comment by arendtio** [comment]
Now that I am back home, I took a look at your comments. I spend the most time on issues that haven't had their own discussion already.

**05:** Your comment is very helpful because I was unaware that these options exist and that they might help with issue 21. However, for 05, they don't help.

I looked closer into it. First, I scrolled down the latest pictures from a few months ago and waited until everything finished
loading. The UI felt fluid as the live photo videos appeared instantly when hovering over a thumbnail with the mouse.
The moment I clicked on a thumbnail, nothing happened for the user (not even the blue loading line at the top), but in the background, the following network request was started:

    https://example.com/api/v1/photos/view?count=9840&offset=0&merged=true&country=&camera=0&lens=0&label=&latlng=&year=0&month=0&color=&order=newest&q=&public=true&quality=3

The request took 10 seconds in that instance (for an 8 MB JSON response), and only after it was completed did the UI switch to the large-picture view. Having such an expensive request between a user interaction and a system response is not a good idea. I don't know under which conditions this request is started, but most of the time, it is not started when switching to the large-picture view. But it becomes larger/more expensive the longer you scroll.

**25:** I rarely use the card view, so I didn't know this link existed there. Thank you for the hint, even though a sidebar for the large-picture view could improve the user experience (in a future version).

**07:** I find it curious that there is already a discussion about 07. However, I do not believe that (in my case) it comes from parallel indexing jobs.

It looked more like some timing issue, with a combination of slow server responses or client processing. My server tends to be slow when ffmpeg converts videos, and my client becomes slow when I browse for too long through months/years of photos (nothing F5 can't fix).

**13, 15, 27:** I'm looking forward to the new viewer. I hope it's as performant as the Photoswipe implementation.

**12:** You are right; the collision with the stack icon isn't that good
either. So, we should choose a different one or change the
stack icon. I looked at the [material icons](https://mui.com/material-ui/material-icons/) and searched for some options. 

Alternatives for the play slide show icon:

- 'Subscriptions'
- VideoLibrary

I don't like the 'slideshow' icon very much, but it has a pause
counterpart (Pause Presentation). For the other icons, I would
just use a regular pause button, as it has a high probability of
being the only pause button on the screen (because slide shows are
meant to be played without interaction, and the videos don't play
automatically at this point).

Another option could be to change the Stack icon. Here are some
options:

- 'layers' (maps & Tensorflow)
- 'RecentActors'
- 'horizontalSplit'
- 'Collection'
- 'SwitchAccount'

Given the collisions with the icon for stacks, my favorite would be the subscriptions icon.

**31, 32:** I agree that they seem to be fixed in the new version

It is good to see that some of the issues I mentioned are already fixed, and many are already being discussed.
